cpp/src/arrow/flight/sql/odbc/odbcabstraction/odbc_impl/odbc_connection.cc

```diff
@@ -696,46 +680,64 @@ void ODBCConnection::dropDescriptor(ODBCDescriptor* desc) {
 
 // Public Static
 // ===================================================================================
-std::string ODBCConnection::getPropertiesFromConnString(
-    const std::string& connStr, Connection::ConnPropertyMap& properties) {
+std::string ODBCConnection::getDsnIfExists(const std::string& connStr) {
   const int groups[] = {1, 2};  // CONNECTION_STR_REGEX has two groups. key: 1, value: 2
-  boost::xpressive::sregex_token_iterator regexIter(connStr.begin(), connStr.end(),
-                                                    CONNECTION_STR_REGEX, groups),
+  boost::xpressive::sregex_token_iterator regex_iter(connStr.begin(), connStr.end(),
+                                                     CONNECTION_STR_REGEX, groups),
       end;
 
-  bool isDsnFirst = false;
-  bool isDriverFirst = false;
-  std::string dsn;
-  for (auto it = regexIter; end != regexIter; ++regexIter) {
-    std::string key = *regexIter;
-    std::string value = *++regexIter;
+  bool dsn_first = false;
+  bool driver_first = false;
+  std::string dsn("");
+  for (auto it = regex_iter; end != regex_iter; ++regex_iter) {
+    std::string key = *regex_iter;
+    std::string value = *++regex_iter;
 
     // If the DSN shows up before driver key, load settings from the DSN.
     // Only load values from the DSN once regardless of how many times the DSN
     // key shows up.
     if (boost::iequals(key, "DSN")) {
```

Comment:
If neither DSN nor DRIVER are the first keys, I think the connection string is faulty. Please check on this as it would simplify this a bit.